### PR TITLE
Repro for issue #13241

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
READY

### What does this PR accomplish
- adds repro for #13241 (already fixed - tests should be passing)
- introduces the hack for cypress' broken `.type()` functionality

### Additional notes:
- ~~at this stage, this test covers the simplest possible scenario~~ ✓
- ~~there is a possibility of adding more custom questions and making assertions more specific~~ ✓
- ~~@flamber if you'd like to see some specific formulas, edge cases, etc, please let me know 🙏~~ ✓

### CODE REVIEW

File: frontend/test/metabase/scenarios/question/custom_column.cy.spec.js [hunk 1]
```diff
@@ -55,4 +57,25 @@ describe("scenarios > question > custom columns", () => {
...
+    popover().within(() => {
+      _typeUsingGet("[contenteditable='true']", customFormula);
+      _typeUsingPlaceholder("Something nice and descriptive", columnName);
+
+      cy.findByText("Done").click();
+    });
+
+    cy.findByText("Visualize").click();
+    // wait for visualization to render (Note: anti-pattern)
+    cy.wait(2000);
```
Comment (dacort) [inline frontend/test/metabase/scenarios/question/custom_column.cy.spec.js:78]: We should definitely avoid `wait`s as much as we can. 

This is probably fine for this one, but we should either wait on `/api/dataset` (example below) or do a `cy.contains` on an element on the results page.

```
      cy.route("POST", "/api/dataset").as("dataset");
      // ... click done
      cy.wait("@dataset");
```

**Review by dacort — APPROVED** [review]
Couple minor change requests. Also please add the `support/Repro` tag.

**Comment by flamber** [comment]
Would be nice with a formula also tests something with columns, like `[Quantity] * 2` and `[Quantity] * [Product.Price]`

**Comment by dacort** [comment]
(in-person comment) Please split up the new custom formula tests so that there is one that repro's #13241 and another one that tests whether "should allow columns in custom formula".

**Review by nemanjaglumac — COMMENTED** [review]
@dacort 
> (in-person comment) Please split up the new custom formula tests so that there is one that repro's #13241 and another one that tests whether "should allow columns in custom formula".

I was just about to push all 3 tests (`forEach()`) inside the `it()` test. I must misunderstood you. So to be absolutely certain, you are saying to have 1 test case `"1 + 1"` that is a direct repro of mentioned issue, and other 2 cases that @flamber proposed in a separate test?

File: frontend/test/__support__/cypress.js [hunk 2]
```diff
@@ -107,6 +107,28 @@ export function typeAndBlurUsingLabel(label, value) {
...
+}
+
+export function _typeUsingPlaceholder(selector, value, delay = 100) {
+  cy.findByPlaceholderText(selector)
+    .click()
+    .type(value, { delay })
+    .clear()
+    .click()
+    .type(value, { delay });
+}
```
[the remark this answers is not in the document]
Comment (dacort) [reply frontend/test/__support__/cypress.js:130]: Yea, totally fine. If we end up reusing these elsewhere, we should move them into a proper helper like `popover` and `modal`.

**Comment by dacort** [comment]
> So to be absolutely certain, you are saying to have 1 test case `"1 + 1"` that is a direct repro of mentioned issue, and other 2 cases that @flamber proposed in a separate test?

Yes. I want a test case that specifically does a simple test to reproduce issue #13241 and then a more generic test case that uses column names. For example, these would be the `it` statements:
`it("can create a custom column metabase#13241")`
`it("can create a custom column with an existing column name")`

File: frontend/test/metabase/scenarios/question/custom_column.cy.spec.js [hunk 3]
```diff
@@ -55,4 +57,25 @@ describe("scenarios > question > custom columns", () => {
...
+    cy.get(".Icon-add_data").click();
+
+    popover().within(() => {
+      _typeUsingGet("[contenteditable='true']", customFormula);
+      _typeUsingPlaceholder("Something nice and descriptive", columnName);
+
+      cy.findByText("Done").click();
+    });
+
+    cy.findByText("Visualize").click();
```
[the remark this answers is not in the document]
Comment (dacort) [reply frontend/test/metabase/scenarios/question/custom_column.cy.spec.js]: Let's just leave the wait for now.

**Comment by nemanjaglumac** [comment]
@dacort - good to merge with these newest changes?

File: frontend/test/metabase/scenarios/question/custom_column.cy.spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/question/custom_column.cy.spec.js:85]: @dacort - updated with waiting for `@dataset` as I started using it in a few other tests as well.

**Review by dacort — APPROVED** [review]
